lineapy/db/db.py

```diff
@@ -751,6 +782,16 @@ def delete_artifact_by_name(
             self.session.delete(res)
         self.renew_session()
 
+    def delete_mlflow_metadata_by_artifact_id(self, artifact_id: int) -> None:
+        """
+        Delete MLflow metadata for the artifact
+        """
+        res_query = self.session.query(MLflowArtifactMetadataORM).filter(
+            MLflowArtifactMetadataORM.artifact_id == artifact_id
+        )
+        res_query.delete()
+        self.renew_session()
+
```

Comment:
This should not be called. For us to be writing to an external system, we should retain every bit of metadata that we can for future audit purposes. Propose adding a delete flag or a status column to indicate that the artifact tied to the specific mlflow's model_uri has been deleted.